.github/workflows/publish.yml

```diff
@@ -96,14 +111,29 @@ jobs:
         run: |
           docker buildx build -f $env:DOCKERFILE -o build .
       - name: upload build
-        if: github.event_name == 'repository_dispatch'
+        if: github.event_name == 'workflow_dispatch'
         shell: bash
         env:
-          RELEASE_TAG: ${{ github.event.client_payload.release_tag }}
+          RELEASE_TAG: ${{ needs.create_release.outputs.release_id }}
           PLATFORM: ${{ matrix.platform }}
           GH_TOKEN: ${{ github.token }}
         working-directory: build
         run: |
           chmod +x "bmx"
           tar -czvf "bmx-$PLATFORM.tar.gz" "bmx"
           gh release upload "$RELEASE_TAG" "bmx-$PLATFORM.tar.gz"
+
+  publish_release:
+    needs: [create_release, build, build_docker]
+    if: |
+      github.event_name == 'workflow_dispatch' && github.ref == 'refs/heads/main'
```

Comment:
if `create_release` already has the correct conditions, we can omit it here
```suggestion
```

Reply:
Good point on that